library/spdm_requester_lib/libspdm_req_communication.c

```diff
@@ -13,6 +14,278 @@ libspdm_return_t libspdm_init_connection(void *spdm_context, bool get_version_on
 
     context = spdm_context;
 
+#if LIBSPDM_FIPS_MODE
+
+#if LIBSPDM_FIPS_TEST_AT_LOAD
+    if (context->crypto_context.tested_algo == 0) {
+        if (!libspdm_fips_run_selftest()) {
+            return LIBSPDM_STATUS_FIPS_FAIL;
+        } else {
+            /*all algos are tested and the all results passed*/
+            context->crypto_context.tested_algo = 0xFF;
+            context->crypto_context.self_test_result = 0xFF;
+        }
+    }
+#endif/*LIBSPDM_FIPS_TEST_AT_LOAD*/
+
+#if LIBSPDM_SHA256_SUPPORT
+    if (((context->crypto_context.tested_algo >>
+          LIBSPDM_FIPS_SELF_TEST_HMAC_SHA256_SHIFT_BIT) & 0x01) == 0) {
+        if (!libspdm_fips_selftest_hmac_sha256()) {
+            context->crypto_context.tested_algo |=
+                (0x01 << LIBSPDM_FIPS_SELF_TEST_HMAC_SHA256_SHIFT_BIT);
+            return LIBSPDM_STATUS_FIPS_FAIL;
```

Comment:
I think it's more logical to always run all selftests, even if one or more algorithms have failed.  At the end, the function would have run all selftests and return overall failed/passed.

Reply:
Yes. In function `bool libspdm_update_fips_selftest_context(void *spdm_context)`, all selftests will be run, even if one algo have failed.